pkg/cloudprovider/provider/baremetal/plugins/tinkerbell/types/types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024 The Machine Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	Staged         string = "Staged"
```

Comment:
A dedicated `type HardwareState string` would help identify what these constants are used for.

Reply:
Honestly I added that then I took out as I actually see no real value other than keep casting the value when comparing it with the API field. Especially the API doesn't know this `HardwareState ` or isn't considering as an Enum.